CIP-1855/CIP-1855.md

```diff
@@ -0,0 +1,94 @@
+---
+CIP: 1855
+Title: Forging policy keys for HD Wallets
+Authors: Samuel Leathers <samuel.leathers@iohk.io>, John Lotoski <john.lotoski@iohk.io>, Michael Bishop <michael.bishop@iohk.io>
+Comments-Summary: Multi-party transaction signing and key management for HD wallets.
+Comments-URI: https://github.com/cardano-foundation/CIPs/wiki/Comments:CIP-1855
+Status: Draft
+Type: Standards
+Created: 2021-06-02
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This document describes how to derive forging policy keys used for minting/burning tokens.
+
+## Glossary
+
+
+Term     | Definition
+---      | ---
+HD       | Hierarchical Deterministic, refers to wallets as described in [BIP-0032].
+
+# Motivation
+
+## Overview 
+
+Forging tokens is derived from a script policy. The script policy includes hashes of keys needed to forge new tokens and must be witnessed by the keys with hashes listed. 
+
+# Specification
+
+## HD Derivation
+
+We consider the following HD derivation paths similarly to [CIP-1852]:
+
+```
+m / purpose' / coin_type' / account_ix' / role / index
+```
+
+
+To associate policy keys to a wallet, we reserve however `purpose=1855'` to reserve for policy keys for forging tokens. The coin type remains `coin_type=1815'` to identify Ada as registered in [SLIP-0044]. We use a hardened index for each policy key as derivation is not needed.
+
+We can summarize the various paths and their respective domain in the following table:
+
+| `purpose` | `coin_type` | `policy_ix`        |
+| ---       | ---         | ---                |
+| `1855'`   | `1815'`     | `[2^31 .. 2^32-1]'` |
+
+
+### Rationale
+
+- ERC20 Converter IOHK is developing needs to keep track of policy keys. Rather than having randomly generated policy keys, a policy key can be associated with a mnemonic which is easier to backup.
+- A 3rd party may want to have multiple tokens tied to same mnemonic, so we allow an index to specify the token.
+
+- We use a different purpose for mainly two reasons:
```

Comment:
I think it would help to also state the reason(s) why _those particular_ derivation indices were chosen, as opposed to other possible derivation indices. And why there are 3 levels of derivation when 2 should work equally well.

Reply:
1855 is just the next number in numerical order.

How do you imagine having only 2 derivation levels? By removing the coin type? I guess it could be removed, but no real advantage either.

Reply:
It seemed a bit redundant to have a second branch for coin type that is always 1815´. Especially because these token policies will be used to mint new coin types.

Reply:
I definitely see your point. For https://github.com/cardano-foundation/CIPs/pull/56 we ended up keeping it

Reply:
@rvl The coin type is really here to prevent clashes across different crypto-currency projects (if someone else uses 1855 for something else for example). While this does not really matter for desktop wallets of our ecosystem like Daedalus who only care about a single crypto, Hardware wallets or generic wallets like Exodus do manipulate multiple currencies and multiple CIP/BIP/TIP/LIP (you name it) proposals.